### PR TITLE
Specify a maximum number of runs to limit the request params -> DoS.

### PR DESCRIPTION
**Pull request by ioquatix**

### CODE REVIEW

**Comment by Sim4n6** [comment]
suppose i set `maximum_runs = profiler_runs+1` and `profiler_runs = 999_999_999` ?

**Comment by ioquatix** [comment]
It's your own fault if you use this code incorrectly. I also think we should document that should avoid using this in production or in any environment where a denial of service would cause an outage.

File: lib/rack/contrib/profiler.rb [hunk 1]
```diff
@@ -35,8 +35,11 @@ def initialize(app, options = {})
       @profile = nil
       @printer = parse_printer(options[:printer] || DEFAULT_PRINTER)
       @times = (options[:times] || 1).to_i
+      @maximum_runs = options.fetch(:maximum_runs, 10)
```
Comment (Sim4n6) [inline lib/rack/contrib/profiler.rb:38]: I'm no good in ruby code, but regarding this line of code can `maximum_runs` come from a user-provided URL parameter?

**Review by Sim4n6 — APPROVED** [review]
LGTM

File: lib/rack/contrib/profiler.rb [hunk 2]
```diff
@@ -61,14 +64,32 @@ def profiling?(env)
...
+      # :maximum option passed to `#initialize`, use the :maximum
+      # option.
+      def runs(request)
+        if profiler_runs = request.params['profiler_runs']
+          profiler_runs = profiler_runs.to_i
+          if profiler_runs > @maximum_runs
+            return @maximum_runs
+          else
+            return profiler_runs
+          end
```
Comment (mpalmer) [inline lib/rack/contrib/profiler.rb:80]: Any reason not to go with this?

```suggestion
          [profiler_runs.to_i, @maximum_runs].min
```

Could also drop the `return`s if you were feeling... implicit.

**Review by mpalmer — APPROVED** [review]

File: lib/rack/contrib/profiler.rb (hunk 2 above)
Comment (ioquatix) [reply lib/rack/contrib/profiler.rb:80]: Without branch coverage, the terse form is tricky to test in a way that gives confidence that every avenue is executed. That being said, we aren't testing coverage... but we should be :)

**Review by jeremyevans — COMMENTED** [review]
I prefer clamp, or at least something similar that sets both a low and high bound.  What are your thoughts?

**Comment by mpalmer** [comment]
I guess in the interests of preventing people from trying to do `"potato"` profile runs, doing `profiler_runs.to_i.clamp(1, @maximum_runs)` would be advisable, but I wouldn't make it a blocker for merge.  The important thing, for this PR, is to prevent the DoS, which the current implementation does.

**Comment by Sim4n6** [comment]
Meanwhile, could you please consider enabling the [private vulnerability reporting](https://docs.github.com/en/code-security/security-advisories/working-with-repository-security-advisories/configuring-private-vulnerability-reporting-for-a-repository) feature for this repo?

**Comment by mpalmer** [comment]
I've enabled that, @Sim4n6, and there's also the option to go through Tidelift for vulnerability disclosure, per the README.

**Comment by ioquatix** [comment]
Okay, I released it, thanks everyone!

**Pull request merged**